app/src/main/java/de/westnordost/streetcomplete/quests/AttachPhotoFragment.java

```diff
@@ -0,0 +1,160 @@
+
+public class AttachPhotoFragment extends Fragment
+{
+	private ArrayList<String> imagePaths = new ArrayList<>();
+	private ArrayList<Bitmap> imageBitmaps = new ArrayList<>();
+
+	File photoFile = null;
+
+	static final int REQUEST_TAKE_PHOTO = 1;
+
+	GridView gridView;
+
+	@Override
+	public View onCreateView(LayoutInflater inflater, ViewGroup container,
+							 Bundle savedInstanceState)
+	{
+		final View panel = inflater.inflate(R.layout.attach_photo_fragment, container, false);
+
+		Button takePhoto = panel.findViewById(R.id.buttonTakeImage);
+		takePhoto.setOnClickListener(new View.OnClickListener()
+		{
+			@Override public void onClick(View v)
+			{
+				takePhoto();
+			}
+		});
+
+		if (!getActivity().getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA))
+		{
+			takePhoto.setVisibility(View.GONE);
+		}
+
+		final NoteImageAdapter noteImageAdapter = new NoteImageAdapter(getActivity(), imageBitmaps);
+		gridView = panel.findViewById(R.id.gridView);
+		gridView.setAdapter(noteImageAdapter);
+		gridView.setOnItemClickListener(new AdapterView.OnItemClickListener()
+		{
+			@Override
+			public void onItemClick(AdapterView<?> parent, View view, final int position, long id)
+			{
+				new AlertDialogBuilder(getActivity())
+						.setMessage(R.string.quest_leave_new_note_photo_delete_title)
+						.setNegativeButton(android.R.string.cancel, null)
+						.setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener()
+						{
+							@Override public void onClick(DialogInterface dialog, int which)
+							{
+								imageBitmaps.remove(position);
+								imagePaths.remove(position);
+								noteImageAdapter.notifyDataSetChanged();
+							}
+						})
+						.show();
+			}
+		});
+
+		return panel;
+	}
+
+	private void takePhoto()
+	{
+		Intent takePhotoIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+		if (takePhotoIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+			try {
+				photoFile = createImageFile();
+			} catch (IOException e) {
+			}
+			if (photoFile != null) {
+				if (Build.VERSION.SDK_INT > 21) {
+					//Use FileProvider for getting the content:// URI, see: https://developer.android.com/training/camera/photobasics.html#TaskPath
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, getUriForFile(getActivity(), "de.westnordost.streetcomplete.fileprovider", photoFile));
+				} else {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+				}
+				startActivityForResult(takePhotoIntent, REQUEST_TAKE_PHOTO);
+			}
+		}
+	}
+
+	@Override
+	public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		if (requestCode == REQUEST_TAKE_PHOTO) {
+			if (resultCode == RESULT_OK)
+			{
+				imageBitmaps.add(ThumbnailUtils.extractThumbnail(BitmapFactory.decodeFile(photoFile.getPath()), 150, 150));
+				imagePaths.add(photoFile.toString());
+
+				LeaveNoteDialog.imagePaths = imagePaths;
+				NoteDiscussionForm.imagePaths = imagePaths;
```

Comment:
This is not the way how it should be done... I know... but I could not find any other method to pass the array with the image paths to the note creation form or get it from there... Could you help me?

Reply:
Sure. Basically, there is an inter-fragment communication in the app already, you can have a look at how it is done by example there: The `MapFragment` has `MapControlsFragment` as a child fragment.

In a nutshell, the parent fragment uses `getChildFragmentManager().findFragmentById(id);` to get its child and then call methods on it. If communication from the child back to the parent is necessary, the parent can register itself as a listener to the child (also done in the mentioned fragments). But I think this is not necessary for this case.

Reply:
I implemented it now the same way how it is done in the ```MapFragment``` but if I call ```getChildFragmentManager().findFragmentById(id);``` the result is ```null``` all the time and I get a ```NullPointerException```

Reply:
sure that you are using the right Id? Is the child fragment manager the right one? (Fragment vs Fragment in support library)

Reply:
This is now fixed